DataProcessing/CryptoCoarseFundamentalUniverseDataDownloader.cs

```diff
@@ -0,0 +1,283 @@
+                {
+                    var coarseByDate = dataByDate[fileName];
+
+                    foreach (var dataTicker in coarseByDate.Keys.ToList())
+                    {
+                        string usdDollorVol;
+
+                        // To USD dollar volume conversion, if <ticker>-USD/BUSD/USDT/USDC/BTC pair exist
+                        if (dataTicker.Substring(dataTicker.Length - 3) != "USD")
+                        {
+                            string refCurrency;
+                            decimal rateToUSD = 1m;
+
+                            if (coarseByDate.ContainsKey($"{baseCurrency[dataTicker]}USD"))
+                            {
+                                refCurrency = $"{baseCurrency[dataTicker]}USD";
+                            }
+                            else if (coarseByDate.ContainsKey($"USD{baseCurrency[dataTicker]}"))
+                            {
+                                refCurrency = $"USD{baseCurrency[dataTicker]}";
+                            }
+                            else if (coarseByDate.ContainsKey($"{baseCurrency[dataTicker]}BUSD"))
+                            {
+                                refCurrency = $"{baseCurrency[dataTicker]}BUSD";
+                            }
+                            else if (coarseByDate.ContainsKey($"BUSD{baseCurrency[dataTicker]}"))
+                            {
+                                refCurrency = $"BUSD{baseCurrency[dataTicker]}";
+                            }
+                            else if (coarseByDate.ContainsKey($"USDT{baseCurrency[dataTicker]}"))
+                            {
+                                rateToUSD = coarseByDate.ContainsKey("USDTUSD") ? 
+                                    decimal.Parse(coarseByDate["USDTUSD"].Split(",").First(), NumberStyles.Any, CultureInfo.InvariantCulture) :
+                                    1m / decimal.Parse(coarseByDate["BUSDUSDT"].Split(",").First(), NumberStyles.Any, CultureInfo.InvariantCulture);
+                                refCurrency = $"USDT{baseCurrency[dataTicker]}";
+                            }
+                            else if (coarseByDate.ContainsKey($"{baseCurrency[dataTicker]}USDC"))
+                            {
+                                rateToUSD = coarseByDate.ContainsKey("USDCUSD") ? 
+                                    decimal.Parse(coarseByDate["USDCUSD"].Split(",").First(), NumberStyles.Any, CultureInfo.InvariantCulture) :
+                                    decimal.Parse(coarseByDate["USDCBUSD"].Split(",").First(), NumberStyles.Any, CultureInfo.InvariantCulture);
+                                refCurrency = $"{baseCurrency[dataTicker]}USDC";
+                            }
+                            else if (coarseByDate.ContainsKey($"{baseCurrency[dataTicker]}BTC"))
+                            {
+                                rateToUSD = coarseByDate.ContainsKey("BTCUSD") ? 
```

Comment:
Interesting, think this section of code should be it's own method with tests 👍, like `GetDollarVolume()`

`BTCUSD` hard coded here is a bit fishy, why is it required, what about, BTCUSDT, etc?